src/transformation/visitors/sourceFile.ts

```diff
@@ -43,3 +50,14 @@ export const transformSourceFileNode: FunctionVisitor<ts.SourceFile> = (node, co
     const trivia = node.getFullText().match(/^#!.*\r?\n/)?.[0] ?? "";
     return lua.createFile(statements, getUsedLuaLibFeatures(context), trivia, node);
 };
+
+function isTopLevelAwait(statement: ts.Statement) {
+    // Check if expression contains an await child, without going into function declarations
+    let containsAwait = false;
+    traverseWithoutCrossingFunction(statement, node => {
```

Comment:
It seems pretty intense to recurse the ast downwards from every statement to find top-level awaits. Why not something along the lines of `findFirstNodeAbove(node, n => isAsyncFunction(n))` in `transformAwaitExpression`?

Reply:
You're right, I updated how this check is done